soc/xtensa/intel_adsp/common/multiprocessing.c

```diff
@@ -70,6 +70,9 @@ __asm__(".section .text.z_soc_mp_asm_entry, \"x\" \n\t"
 	"  movi  a0, 0x4002f        \n\t" /* WOE | UM | INTLEVEL(max) */
 	"  wsr   a0, PS             \n\t"
 	"  movi  a0, 0              \n\t"
+#ifdef XCHAL_HAVE_THREADPTR
+	"  wur   a0, THREADPTR      \n\t"
+#endif
```

Comment:
...like here, where we have cut and pasted that initialization block into the platform code (honestly this may have been my doing, I can't remember) instead of unifying it across CPUs.  We should have a single, clean "start here to initialize an Xtensa CPU" path.

Reply:
that is a good point, I got myself having problems after changing for core 0 and the code not working properly on SMP because was missing here. Still that is another problem.
Will take a look on it.